backend/src/modules/match-module/infrastructure/repository/mongo/MongoMatchRepository.ts

```diff
@@ -0,0 +1,60 @@
+import { MatchRepository } from '../../../core/application/MatchRepository';
+import { MatchId } from '../../../core/domain/MatchId';
+import { Match } from '../../../core/domain/Match';
+import mongoose, { Schema } from 'mongoose';
+import { MatchSideId } from '../../../core/domain/MatchSideId';
+
+export class MongoMatchRepository implements MatchRepository {
+  async findByMatchId(matchId: MatchId): Promise<Match | undefined> {
+    const mongoFindResult = await MongoMatch.findById(matchId.raw);
+    return mongoFindResult ? mongoDocumentToDomain(mongoFindResult) : undefined;
+  }
+
+  async findAll(): Promise<Match[]> {
+    const mongoFindResult = await MongoMatch.find();
+    return mongoFindResult.map((mongoDocument) => mongoDocumentToDomain(mongoDocument));
+  }
+
+  async save(match: Match): Promise<void> {
+    await MongoMatch.findByIdAndUpdate(
+      { _id: match.matchId.raw },
+      {
+        _id: match.matchId.raw,
+        firstMatchSideId: match.firstMatchSideId.raw,
+        secondMatchSideId: match.secondMatchSideId.raw,
+      },
+      { upsert: true, useFindAndModify: true },
+    );
+  }
+}
+
+type MongoMatch = {
+  readonly _id: string;
+  readonly firstMatchSideId: string;
+  readonly secondMatchSideId: string;
+  readonly winner: string;
```

Comment:
Jak dla mnie pola: looser i hasEnded są nadmiarowe. Skoro jest winner, to wiadomo, że drugi jest looser. A skoro mamy winner, to znaczy, że mecz się skończył :) Na poziomie modelu domenowego możesz trzymać te pola, jeśli są potrzebne do jakiejś logiki. Albo zaimplementować to jako gettery.

Reply:
W sumie to dodałeś te pola, a one w ogóle nie są jak na razie potrzebne. Tzn. będą, ale nie implementowałeś jeszcze kwestii związanych z kończeniem meczu :) Tzn. całej komendy zapisania wyniku itp.

Reply:
Tak samo jak sobie mysle, na turnieju zglasza sie zazwyczaj "kto wygrał". No bo cała reszta już się rozumie sama przez się :)

Reply:
przyznaję, że trochę wyprzedziłem zakres tego zadania :) jeśli chodzi o pole looser to z Anią rozmawialiśmy, że przegrany też ma jeszcze szansę w turnieju bo odpada na 'lewo', więc uznałem, że taki pole może się przydać :) a co do hasEnded to może faktycznie nadmiarowe i możemy polegać na winner (jakoś o tym nie pomyślałem :P )

Reply:
Proponuje skasowac z Match pola looser, hasEnded. 
Masz tez metody :) 

Ale loosera tez mozesz latwo dostac:
```
{
  get looser(): MatchSideId | undefined {
      return ....
  }

get hasEnded(): boolean {
   return this.winner !== undefined 
 }
}
```

I nie podajesz ich wtedy nigdy tylko "wyliczasz".

Reply:
A w bazie danych wtedy w ogole tego nie masz. Ona sluzy Ci tylko do tego, zebys mógł poprawnie wykonać proces, który ma więcej niż 1 krok (czyli musisz cos w miedzyczasie zapisać).